### PR TITLE
Adding StringUtils class and test file.

### PR DESCRIPTION
**Pull request by Anshika1603**
Please review and allow me to merge JUnit4 Assignment.

### CODE REVIEW

File: src/main/java/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import java.util.Scanner;
+
+class StringUtils {
+
+    // Method to reverse the String
+    public String reverse(String inputString){
+        if(isEmpty(inputString)) {
+            return "Invalid Input";
+        }
+        else {
```
Comment (anup0060) [inline src/main/java/StringUtils.java]: if first if returning , then no need to have else.

File: src/main/java/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            if (inputString.equals(reversed)) {
+                return true;
+            } else
+                return false;
+        }
+    }
+
+    public static void main(String[] args) {
+        StringUtils stringUtils=new StringUtils();
+        Scanner Input=new Scanner(System.in);
```
Comment (anup0060) [inline src/main/java/StringUtils.java]: poor variable naming .

File: src/main/java/StringUtils.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        System.out.println("Enter the String: ");
+        String UserInput=Input.nextLine(); // Taking input from user
```
Comment (anup0060) [inline src/main/java/StringUtils.java]: use camel casing for variables

File: src/test/java/StringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+public class StringUtilsTest {
+    StringUtils stringUtils;
+    @Before
+    public void setup(){
+        stringUtils=new StringUtils();
+    }
+
+    @Test
+    public void testReverse_shouldReturn_String(){
```
Comment (anup0060) [inline src/test/java/StringUtilsTest.java]: you also need to mention given. means when should is expected. 

testReverse_shouldReturnString_givenExpectedInput

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Comment by Anshika1603** [comment]
I have worked on the changes and updated the classes. Please review it.

File: src/test/java/StringOperationsTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Test
+    public void testReverse_shouldReturnString_givenString(){
+        assertEquals("cba", stringOperations.reverse("abc"));
+        assertEquals("Invalid Input", stringOperations.reverse(""));
+        assertEquals("racecar", stringOperations.reverse("racecar"));
+    }
+
+    @Test
+    public void testIsEmpty_shouldReturnBoolean_givenString(){
```
Comment (anup0060) [inline src/test/java/StringOperationsTest.java]: empty case should be separate (we can have case for true and other for false)

File: src/test/java/StringOperationsTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        assertTrue(stringOperations.isEmpty(""));
+        assertFalse(stringOperations.isEmpty("hello"));
+    }
+
+    @Test
+    public void testIsPalindrome_shouldReturnBoolean_givenString(){
+        assertTrue(stringOperations.isPalindrome("racecar"));
+        assertTrue(stringOperations.isPalindrome("madam"));
+        assertFalse(stringOperations.isPalindrome("hello"));
+        assertFalse(stringOperations.isPalindrome(""));
```
Comment (anup0060) [inline src/test/java/StringOperationsTest.java:43]: empty case separate

File: src/main/java/StringOperations.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import java.util.Scanner;
+
+class StringOperations {
+
+    // Method to reverse the String
+    public String reverse(String inputStatement){
```
Comment (anup0060) [inline src/main/java/StringOperations.java]: variable name inputStatement not good. you can say - inputStr

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Comment by Anshika1603** [comment]
I have added different test methods for empty cases and also changed the variable name. Please review.

**Review by anup0060 — APPROVED** [review]

**Pull request merged**